### ISSUE TITLE
Tree functionnalities broken

### ISSUE THREAD

**Issue by thonythony** [typeorm/typeorm#3783]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

All find* tree functionnalities in `TreeRepository` are broken due to this commit (following to support of cockroachdb) when primary keys are uuid. This commit tries to convert an uuid to integer, breaking the relation mechanism.

https://github.com/typeorm/typeorm/blob/a12e72d417638ea8fba2ca672d5acfaaaaf691c9/src/repository/TreeRepository.ts#L258
https://github.com/typeorm/typeorm/blob/a12e72d417638ea8fba2ca672d5acfaaaaf691c9/src/repository/TreeRepository.ts#L259

**Comment by thonythony** [comment]
@pleerock

**Comment by pleerock** [comment]
We are working on a fix

**Comment by thonythony** [comment]
I have this fix that I can contribute : 
`TreeRepository.ts`
```typescript
 protected createRelationMaps(alias: string, rawResults: any[]): { id: any, parentId: any }[] {
        return rawResults.map(rawResult => {
            const joinColumn = this.metadata.treeParentRelation!.joinColumns[0];
            // fixes issue #2518, default to databaseName property when givenDatabaseName is not set
            const joinColumnName = joinColumn.givenDatabaseName || joinColumn.databaseName;
            let id = rawResult[alias + "_" + this.metadata.primaryColumns[0].databaseName];
            let parentId = rawResult[alias + "_" + joinColumnName];
            // CockroachDB returns numeric types as string
            const uuidPattern = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}$/i;

            if (uuidPattern.test(id) === false) {
                const idParsed = parseInt(id);
                if (isNaN(idParsed) === false) {
                    id = idParsed;
                }
            }

            if (uuidPattern.test(parentId) === false) {
                const parentIdParsed = parseInt(parentId);
            
                if (isNaN(parentIdParsed) === false) {
                    parentId = parentIdParsed;
                }
            }
            
            return {
                id,
                parentId
            };
        });
    }
```

**Comment by pleerock** [comment]
Fix was released in `0.2.15-rc.2`
